controller/controller.go

```diff
@@ -97,60 +101,41 @@ func (c *Controller) Login(ctx *gin.Context) {
 
 // @Tags timer
 // @Summary Start timer
-// @param X-AUTH-TOKEN	header	string	true	"JWT Token"
+// @param X-Auth-Token	header	string	true	"JWT Token"
 // @Accept  json
 // @Produce  json
-// @router /start [post]
-// @Success 200 {object} model.Entry
-func (c *Controller) StartTime(ctx *gin.Context) {
-
-	token := ctx.GetHeader("X-AUTH-TOKEN")
-	if len(token) == 0 {
-		ctx.JSON(http.StatusBadRequest, "token is missing")
-		return
-	}
+// @router /tracker/start [post]
+// @Success 200 {object} models.Entry
+func (ctrl *Controller) StartTimePost(ctx *gin.Context) {
 
-	id, err := utils.ValidateToken(token)
-	if err != nil {
-		ctx.JSON(http.StatusBadRequest, err)
-		return
-	}
+	userIdString, _ := ctx.Get(utils.UserId)
+	userId := userIdString.(int)
 
-	entry, err := c.DB.AddTimeEntry(id)
+	entry, err := ctrl.db.AddTimeEntry(userId)
 	if err != nil {
 		ctx.JSON(http.StatusInternalServerError, err)
 		return
 	}
-	entry.UserId = id
+	entry.UserId = userId
 	ctx.JSON(http.StatusOK, entry)
 }
 
 // @Tags timer
 // @Summary Stop timer
 // @Param id path	string 	true 	"id of the entry"
-// @param X-AUTH-TOKEN	header	string	true	"JWT Token"
+// @param X-Auth-Token	header	string	true	"JWT Token"
 // @Accept  json
 // @Produce  json
-// @router /stop/{id} [put]
+// @router /tracker/stop/{id} [put]
 // @Success 200 "record updated"
-func (c *Controller) StopTime(ctx *gin.Context) {
+func (ctrl *Controller) StopTimePut(ctx *gin.Context) {
 
-	token := ctx.GetHeader("X-AUTH-TOKEN")
-	if len(token) == 0 {
-		ctx.JSON(http.StatusBadRequest, "token is missing")
-		return
-	}
-
-	user_id, err := utils.ValidateToken(token)
-	if err != nil {
-		ctx.JSON(http.StatusBadRequest, err)
-		return
-	}
+	userIdString, _ := ctx.Get(utils.UserId)
+	userId := userIdString.(int)
```

Comment:
This isn't the proper way to typecast, the code will break if userIdString cannot be converted to int